UGI-216004-main/PhotoEnhancer/PhotoEnhancer/Program.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Security.Cryptography;
+
+namespace PhotoEnhancer
+{
+    internal static class Program
+    {
+        /// <summary>
+        /// Главная точка входа для приложения.
+        /// </summary>
+        [STAThread]
+        static void Main()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+
+            var mainForm = new MainForm();
+
+            mainForm.AddFilter(new PixelFilter<LighteningParameters>(
+                "Осветление/затемнение",
+                (pixel, parameters) => pixel * parameters.Coefficient));
+            
+            mainForm.AddFilter(new PixelFilter<EmptyParameters>(
+                "Оттенки серого",
+                (pixel, parameters) =>
+                {
+                    var lightness = 0.3 * pixel.R + 0.6 * pixel.G + 0.1 * pixel.B;
+                    return new Pixel(lightness, lightness, lightness);
+                }));
+
+            mainForm.AddFilter(new PixelFilter<NoiseParameters>(
+                "Шум",
+                (pixel, parameters) =>
+                {
+                    Random rnd = new Random();
```

Comment:
А фильтр шума вы так и не исправили. А в задании нужно было, чтобы все фильтры работали. А всего-то нудно было вынести из делегата переменную (организовать замыкание).
```suggestion
            Random rnd = new Random();

            mainForm.AddFilter(new PixelFilter<NoiseParameters>(
                "Шум",
                (pixel, parameters) =>
                {
```